purchases/src/main/kotlin/com/revenuecat/purchases/models/TestStoreProduct.kt

```diff
@@ -0,0 +1,95 @@
+package com.revenuecat.purchases.models
+
+import com.revenuecat.purchases.ProductType
+
+/**
+ * A test-only [StoreProduct] implementation.
+ * This can be used to create mock data for tests or Jetpack Compose previews.
+ */
+data class TestStoreProduct(
+    override val id: String,
+    override val title: String,
+    override val description: String,
+    override val price: Price,
+    override val period: Period?,
+    private val freeTrialPeriod: Period? = null,
+    private val introPrice: Price? = null,
+) : StoreProduct {
+    override val type: ProductType
+        get() = if (period == null) ProductType.INAPP else ProductType.SUBS
+    override val subscriptionOptions: SubscriptionOptions?
+        get() = buildSubscriptionOptions()
+    override val defaultOption: SubscriptionOption?
+        get() = subscriptionOptions?.defaultOffer
+    override val purchasingData: PurchasingData
+        get() = object : PurchasingData {
+            override val productId: String
+                get() = id
+            override val productType: ProductType
+                get() = type
```

Comment:
This was wrong in the existing implementation, now correctly derived from the above logic